index.go

```diff
@@ -65,6 +65,18 @@ type EventIndex interface {
 	FireIndexEvent()
 }
 
+type UpdateFieldInfo struct {
+	RemoveAll bool
```

Comment:
This looks quite unintuitive as well @Likith101  - perhaps some commentary around each would help.

Reply:
Simply change to `Remove` maybe.